datajoint/base_relation.py

```diff
@@ -171,106 +171,153 @@ def full_table_name(self):
         :param ignore_errors=False: If True, ignore errors: e.g. constraint violations.
         :param skip_dublicates=False: If True, silently skip duplicate inserts.
 
         Example::
 
-        >>> relation.insert1(dict(subject_id=7, species="mouse", date_of_birth="2014-09-01"))
+        >>> relation.insert([
+        >>>     dict(subject_id=7, species="mouse", date_of_birth="2014-09-01"),
+        >>>     dict(subject_id=8, species="mouse", date_of_birth="2014-09-02")])
 
         """
-
         heading = self.heading
+        field_list = None
 
-        def check_fields(fields):
+        def make_row_to_insert(row):
             """
-            Validates that all items in `fields` are valid attributes in the heading
+            :param row:  A tuple to insert
+            :return: a dict with fields 'names', 'placeholders', 'values'
             """
-            for field in fields:
-                if field not in heading:
-                    raise KeyError(u'{0:s} is not in the attribute list'.format(field))
 
-        def make_attribute(name, value):
+            def make_placeholder(name, value):
+                """
+                For a given attribute `name` with `value, return its processed value or value placeholder
+                as a string to be included in the query and the value, if any to be submitted for
+                processing by mysql API.
+                :param name:
+                :param value:
+                """
+                if heading[name].is_blob:
+                    value = pack(value)
+                    # This is a temporary hack to address issue #131 (slow blob inserts).
+                    # When this problem is fixed by pymysql or python, then pass blob as query argument.
+                    placeholder = '0x' + binascii.b2a_hex(value).decode('ascii')
+                    value = None
+                elif heading[name].numeric:
+                    if value is None or np.isnan(value):   # nans are turned into NULLs
+                        placeholder = 'NULL'
+                        value = None
+                    else:
+                        placeholder = '%s'
+                        value = repr(int(value) if isinstance(value, bool) else value)
+                else:
+                    placeholder = '%s'
+                return name, placeholder, value
+
+            def check_fields(fields):
+                    """
+                    Validates that all items in `fields` are valid attributes in the heading
+                    :param fields: field names of a tuple
+                    """
+                    if field_list is None:
+                        for field in fields:
+                            if field not in heading:
+                                raise KeyError(u'{0:s} is not in the attribute list'.format(field))
+                    else:
+                        if len(fields) < len(field_list):
```

Comment:
When inserting a list of tuples, all tuples must have the same columns.  That's what is required by the SQL syntax.  If the insert relies on default values, then all tuples must rely on default values, not just some.